php/assets/class-rest-assets.php

```diff
@@ -105,30 +95,72 @@ public function rest_endpoints( $endpoints ) {
 	 */
 	public function rest_purge( $request ) {
 
-		$asset      = $request->get_param( 'cachePoint' );
-		$query_args = array(
-			'post_type'      => Assets::POST_TYPE_SLUG,
-			'posts_per_page' => 20,
-			'paged'          => 1,
-			'post_parent'    => $asset,
-			'post_status'    => array( 'inherit', 'draft' ),
-			'fields'         => 'ids',
-		);
-		$query      = new \WP_Query( $query_args );
-		do {
-			$posts = $query->get_posts();
-			foreach ( $posts as $post_id ) {
-				wp_delete_post( $post_id );
+		$asset_parent  = (int) $request->get_param( 'asset_parent' );
+		$transient_key = '_purge_cache' . $asset_parent;
+		$parents       = $this->assets->get_asset_parents();
+		if ( empty( $parents ) ) {
+			return rest_ensure_response( true );
+		}
+
+		$tracker = get_transient( $transient_key );
+		foreach ( $parents as $parent ) {
+			if ( $asset_parent && $asset_parent !== $parent->ID ) {
+				continue;
 			}
-			// Paginate.
-			$query_args = $query->query_vars;
-			$query_args['paged'] ++;
-			$query = new \WP_Query( $query_args );
-		} while ( $query->have_posts() );
+			$tracker['time']           = time();
+			$tracker['current_parent'] = $asset_parent;
+			set_transient( $transient_key, $tracker, MINUTE_IN_SECONDS );
+			$this->assets->purge_parent( $parent->ID );
+		}
+		delete_transient( $transient_key );
+
+		return rest_ensure_response( true );
+	}
 
-		$deleted = wp_delete_post( $asset );
+	/**
+	 * Purges a cache which forces the entire point to re-evaluate cached items when requested.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_Error|WP_HTTP_Response|WP_REST_Response
+	 */
+	public function rest_purge_all( $request ) {
+
+		$count         = $request->get_param( 'count' );
+		$asset_parent  = (int) $request->get_param( 'parent' );
+		$transient_key = '_purge_cache' . $asset_parent;
+		$query_args    = array(
+			'post_type'              => Assets::POST_TYPE_SLUG,
+			'posts_per_page'         => 1,
+			'paged'                  => 1,
+			'post_status'            => array( 'inherit', 'draft', 'publish' ),
```

Comment:
@DavidCramer 

I'm a bit confused here. What's does the `publish` state stands for?
I thought that we only had 2 states as per this [comment](https://github.com/cloudinary/cloudinary_wordpress/pull/484/files#diff-12a132c16918f2f8786165885c22904fba05f2c21467113a4fda1dbeed882458R164-R165).

Reply:
Publish is ignored withing the media post types, so publish is given to the parent post.